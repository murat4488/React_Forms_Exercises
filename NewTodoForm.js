import React, { useState } from "react";

const NewTodoForm = ({ createTodo }) => {
  const INITIAL_STATE = { text: "" };

  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleSubmit = (e) => {
    e.preventDefault();
    createTodo({ ...formData });
    setFormData(INITIAL_STATE);
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="add"></label>
      <input
        type="text"
        name="text"
        placeholder="add"
        value={formData.text}
        onChange={handleChange}
      />
      <button className="todo-form" w type="submit">
        Add
      </button>
    </form>
  );
};
export default NewTodoForm;
