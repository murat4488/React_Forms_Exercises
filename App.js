import { useState } from "react";
import "./App.css";
import BoxList from "./BoxList";
import NewBoxForm from "./NewBoxForm";
import TodoList from "./TodoList";
import NewTodoForm from "./NewTodoForm";

function App() {
  const [boxes, setBoxes] = useState([]);

  const createBox = ({ color, width, height }) => {
    const newBox = { id: Date.now(), color, width, height };
    setBoxes((boxes) => [...boxes, newBox]);
  };

  const removeBox = (boxId) => {
    const updatedBoxes = boxes.filter((box) => boxId !== box.id);
    setBoxes(updatedBoxes);
  };

  const [todos, setTodos] = useState([]);

  const createTodo = ({ text }) => {
    const newTodo = { id: Date.now(), text };
    setTodos((todos) => [...todos, newTodo]);
  };

  const removeTodo = (todoId) => {
    const updatedTodos = todos.filter((todo) => todoId !== todo.id);
    setTodos(updatedTodos);
  };

  return (
    <div className="App">
      <div className="box-form">
        <h1>Create Box</h1>
        <NewBoxForm createBox={createBox} />
        <BoxList boxes={boxes} removeBox={removeBox} />
        <h1>Create Todos</h1>
        <NewTodoForm createTodo={createTodo} />
        <TodoList todos={todos} removeTodo={removeTodo} />
      </div>
    </div>
  );
}

export default App;
