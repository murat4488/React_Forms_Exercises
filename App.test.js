import { render, fireEvent } from "@testing-library/react";
import App from "./App";
import NewTodoForm from "./NewTodoForm";

test("smoke test", () => {
  render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();
});

test("matches snapshot", function () {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
