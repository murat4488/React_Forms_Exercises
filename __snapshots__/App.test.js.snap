// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="box-form"
    >
      <h1>
        Create Box
      </h1>
      <form>
        <label
          for="color"
        >
          Color
        </label>
        <input
          name="color"
          placeholder="color"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          name="width"
          placeholder="width"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height
        </label>
        <input
          name="height"
          placeholder="height"
          type="text"
          value=""
        />
        <button
          class="new-box-form"
          type="submit"
        >
          Add New Box
        </button>
      </form>
      <div />
      <h1>
        Create Todos
      </h1>
      <form>
        <label
          for="add"
        />
        <input
          name="text"
          placeholder="add"
          type="text"
          value=""
        />
        <button
          class="todo-form"
          type="submit"
        >
          Add
        </button>
      </form>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
